### ISSUE TITLE
Cannot view contacts v 4.1.2- SOLVED

### ISSUE THREAD

**Issue by dinaras** [mautic/mautic#10885]
### Mautic Version

4.1.x series

### PHP version

7.4.25

### What browsers are you seeing the problem on?

Not relevant

### What happened?

i expect when i click contacts that i can see contacts, but know it shows that it is loading and does nothing.

The only similar post i have found is the following but i cannot apply the solution to my version as files have been already changed. Apart from that i could not find any more info

This is not a clean install, we have applied upgrades a couple times at least as we use mautic the last few months

Thank you in advance for your help
Regards
Dinos

### How can we reproduce this issue?

Step 1: login to your mautic installation
Step 2: Click contacts
Step 3: Click a Contact to view its details

Nothing will show

### Relevant log output

```shell
[2022-02-22 12:10:09] mautic.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\DriverException: "An exception occurred while executing 'SELECT ip.date_added, ip.ip_address, ip.lead_id, ip.id FROM (SELECT MAX(l.date_added) as date_added, MIN(l.id) as id, l.ip_address, l.object_id as lead_id FROM audit_log l WHERE (l.bundle = 'lead') AND (l.object = 'lead') AND (l.action = 'ipadded') AND ((l.object_id = 28877) AND (l.date_added >= '2022-01-25 12:01:46')) GROUP BY l.ip_address) ip ORDER BY ip.date_added DESC LIMIT 25':  SQLSTATE[42000]: Syntax error or access violation: 1055 'mautic.l.object_id' isn't in GROUP BY" at /vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 128 {"exception":"[object] (Doctrine\\DBAL\\Exception\\DriverException(code: 0): An exception occurred while executing 'SELECT ip.date_added, ip.ip_address, ip.lead_id, ip.id FROM (SELECT MAX(l.date_added) as date_added, MIN(l.id) as id, l.ip_address, l.object_id as lead_id FROM audit_log l WHERE (l.bundle = 'lead') AND (l.object = 'lead') AND (l.action = 'ipadded') AND ((l.object_id = 28877) AND (l.date_added >= '2022-01-25 12:01:46')) GROUP BY l.ip_address) ip ORDER BY ip.date_added DESC LIMIT 25':\n\nSQLSTATE[42000]: Syntax error or access violation: 1055 'mautic.l.object_id' isn't in GROUP BY at /vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:128, Doctrine\\DBAL\\Driver\\PDO\\Exception(code: 42000): SQLSTATE[42000]: Syntax error or access violation: 1055 'mautic.l.object_id' isn't in GROUP BY at /vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDO/Exception.php:18, PDOException(code: 42000): SQLSTATE[42000]: Syntax error or access violation: 1055 'mautic.l.object_id' isn't in GROUP BY at /vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:132)"} []
```

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

**Comment by dinaras** [comment]
From the server i can see
192.x.x.x - - [22/Feb/2022:12:34:43 +0200] "GET /s/contacts/view/39799?_=1645517810339&mauticUserLastActive=8274&mauticLastNotificationId= HTTP/2.0" 500 161 "https:xxxxxx/s/contacts" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.80 Safari/537.36" "-"

thanks again for your hard work
Dinos

**Comment by Aerendir** [comment]
Very probably something went wrong with your update.

Via FTP, go to `var/logs`. Here you'll see all the logs wrote by Mautic.

Open the most recent and see what it says: there you will find the cause of the 500 error.

**Comment by dinaras** [comment]
Hello
Following your suggestion i managed to find the following from /var/log/nginx/ error logs


`2022/02/22 12:41:47 [error] 32445#0: *1398256 FastCGI sent in stderr: "PHP message: PHP Notice:  Undefined variable: columns in app/bundles/LeadBundle/Views/Lead/list_rows.html.php on line 85PHP message: PHP Warning:  array_flip() expects parameter 1 to be array, null given in app/bundles/LeadBundle/Views/Lead/list_rows.html.php on line 85PHP message: PHP Notice:  Undefined variable: columns in app/bundles/LeadBundle/Views/Lead/list_rows.html.php on line 86PHP message: PHP Warning:  Invalid argument supplied for foreach() in app/bundles/LeadBundle/Views/Lead/list_rows.html.php on line 86PHP message: PHP Notice:  Undefined variable: columns in app/bundles/LeadBundle/Views/Lead/list_rows.html.php on line 85PHP message: PHP Warning:  array_flip() expects parameter 1 to be array, null given in app/bundles/LeadBundle/Views/Lead/list_rows.html.php on line 85PHP message: PHP Notice:  Undefined variable: columns in app/bundles/LeadBundle/Views/Lead/list_rows.html.php on line 86PHP message: PHP Warning:  Invalid argument supplied for foreach() in app/bundles/LeadBundle/Views/Lead/list_rows.html.php on line 86PHP message: PHP Notice:  Undefined variable: columns in app/bundles/LeadBundle/Views/Lead/list_rows.html.php on line 85PHP message: PHP Warning:  array_flip() expects parameter 1 to be array, null given in app/bundles/LeadBundle/Views/Lead/list_rows.html.php on line 85PHP message: PHP Notice:  Unde
`

Also from Mautic logs is found the following errors also

`[2022-02-22 13:07:02] mautic.ERROR: Failed to fetch remote IP data: URI must be a string or UriInterface [] [].`
The above is when trying to view anonymous ip details

and finally from mautic logs again

`[2022-02-22 12:41:47] mautic.WARNING: PHP Warning - Invalid argument supplied for foreach() - in file app/bundles/LeadBundle/Views/Lead/list_rows.html.php - at line 86 {"mauticTemplateVars":{"app":"[object] (Symfony\\Bundle\\FrameworkBundle\\Templating\\GlobalVariables: {})","items":{"42285":"[object] (Mautic\\LeadBundle\\Entity\\Lead: Mautic\\LeadBundle\\Entity\\Lead with ID #42285)","42284":"[object] (Mautic\\LeadBundle\\Entity\\Lead: Mautic\\LeadBundle\\Entity\\Lead with ID #42284)","42283":"[object] (Mautic\\LeadBundle\\Entity\\Lead: Mautic\\LeadBundle\\Entity\\Lead with ID #42283)","42282":"[object] (Mautic\\LeadBundle\\Entity\\Lead: Mautic\\LeadBundle\\Entity\\Lead with ID #42282)","42281":"[object] (Mautic\\LeadBundle\\Entity\\Lead: Mautic\\LeadBundle\\Entity\\Lead with ID #42281)","42280":"[object] (Mautic\\LeadBundle\\Entity\\Lead: Mautic\\LeadBundle\\Entity\\Lead with ID #42280)","42279":"[object] (Mautic\\LeadBundle\\Entity\\Lead: Mautic\\LeadBundle\\Entity\\Lead with ID #42279)"},"noContactList":[],"permissions":{"lead:leads:viewown":true,"lead:leads:viewother":true,"lead:leads:create":true,"lead:leads:editown":true,"lead:leads:editother":true,"lead:leads:deleteown":true,"lead:leads:deleteother":true},"security":"[object] (Mautic\\CoreBundle\\Security\\Permissions\\CorePermissions: {})","highlight":true,"mauticTemplate":"MauticLeadBundle:Lead:list_rows.html.php","view":"[object] (Mautic\\CoreBundle\\Templating\\Engine\\PhpEngine: {})"},"app":"[object] (Symfony\\Bundle\\FrameworkBundle\\Templating\\GlobalVariables: {})","items":{"42285":"[object] (Mautic\\LeadBundle\\Entity\\Lead: Mautic\\LeadBundle\\Entity\\Lead with ID #42285)","42284":"[object] (Mautic\\LeadBundle\\Entity\\Lead: Mautic\\LeadBundle\\Entity\\Lead with ID #42284)","42283":"[object] (Mautic\\LeadBundle\\Entity\\Lead: Mautic\\LeadBundle\\Entity\\Lead with ID #42283)","42282":"[object] (Mautic\\LeadBundle\\Entity\\Lead: Mautic\\LeadBundle\\Entity\\Lead with ID #42282)","42281":"[object] (Mautic\\LeadBundle\\Entity\\Lead: Mautic\\LeadBundle\\Entity\\Lead with ID #42281)","42280":"[object] (Mautic\\LeadBundle\\Entity\\Lead: Mautic\\LeadBundle\\Entity\\Lead with ID #42280)","42279":"[object] (Mautic\\LeadBundle\\Entity\\Lead: Mautic\\LeadBundle\\Entity\\Lead with ID #42279)"},"noContactList":[],"permissions":{"lead:leads:viewown":true,"lead:leads:viewother":true,"lead:leads:create":true,"lead:leads:editown":true,"lead:leads:editother":true,"lead:leads:deleteown":true,"lead:leads:deleteother":true},"security":"[object] (Mautic\\CoreBundle\\Security\\Permissions\\CorePermissions: {})","highlight":true,"mauticTemplate":"MauticLeadBundle:Lead:list_rows.html.php","view":"[object] (Mautic\\CoreBundle\\Templating\\Engine\\PhpEngine: {})","item":"[object] (Mautic\\LeadBundle\\Entity\\Lead: Mautic\\LeadBundle\\Entity\\Lead with ID #42279)","fields":{"core":{"points":{"id":"9","label":"Points","alias":"points","type":"number","group":"core","object":"lead","is_fixed":"1","properties":"a:0:{}","default_value":"0","value":"0","normalizedValue":"0"},"last_active":{"id":"19","label":"Date Last Active","alias":"last_active","type":"datetime","group":"core","object":"lead","is_fixed":"1","properties":"a:0:{}","default_value":null,"value":"2022-02-22 10:39:24","normalizedValue":"2022-02-22 10:39:24"},"title":{"id":"1","label":"Title","alias":"title","type":"lookup","group":"core","object":"lead","is_fixed":"1","properties":"a:1:{s:4:\"list\";a:3:{i:0;s:2:\"Mr\";i:1;s:3:\"Mrs\";i:2;s:4:\"Miss\";}}","default_value":null,"value":null,"normalizedValue":null},"firstname":{"id":"2","label":"First Name","alias":"firstname","type":"text","group":"core","object":"lead","is_fixed":"1","properties":"a:0:{}","default_value":null,"value":null,"normalizedValue":null},"lastname":{"id":"3","label":"Last Name","alias":"lastname","type":"text","group":"core","object":"lead","is_fixed":"1","properties":"a:0:{}","default_value":null,"value":null,"normalizedValue":null},"company":{"id":"4","label":"Primary company","alias":"company","type":"text","group":"core","object":"lead","is_fixed":"1","properties":"a:0:{}","default_value":null,"value":null,"normalizedValue":null},"position":{"id":"5","label":"Position","alias":"position","type":"text","group":"core","object":"lead","is_fixed":"1","properties":"a:0:{}","default_value":null,"value":null,"normalizedValue":null},"email":{"id":"6","label":"Email","alias":"email","type":"email","group":"core","object":"lead","is_fixed":"1","properties":"a:0:{}","default_value":null,"value":null,"normalizedValue":null},"phone":{"id":"8","label":"Phone","alias":"phone","type":"tel","group":"core","object":"lead","is_fixed":"1","properties":"a:0:{}","default_value":null,"value":null,"normalizedValue":null},"mobile":{"id":"7","label":"Mobile","alias":"mobile","type":"tel","group":"core","object":"lead","is_fixed":"1","properties":"a:0:{}","default_value":null,"value":null,"normalizedValue":null},"address1":{"id":"11","label":"Address Line 1","alias":"address1","type":"text","group":"core","object":"lead","is_fixed":"1","properties":"a:0:{}","default_value":null,"value":null,"normalizedValue":null},"address2":{"id":"12","label":"Address Line 2","alias":"address2","type":"text","group":"core","object":"lead","is_fixed":"1","properties":"a:0:{}","default_value":null,"value":null,"normalizedValue":null},"city":{"id":"13","label":"City","alias":"city","type":"text","group":"core","object":"lead","is_fixed":"1","properties":"a:0:{}","default_value":null,"value":"Athens","normalizedValue":"Athens"},"state":{"id":"14","label":"State","alias":"state","type":"region","group":"core","object":"lead","is_fixed":"1","properties":"a:0:{}","default_value":null,"value":"Attica","normalizedValue":"Attica"},"zipcode":{"id":"15","label":"Zip Code","alias":"zipcode","type":"text","group":"core","object":"lead","is_fixed":"1","properties":"a:0:{}","default_value":null,"value":null,"normalizedValue":null},"timezone":{"id":"18","label":"Preferred Timezone","alias":"timezone","type":"timezone","group":"core","object":"lead","is_fixed":"1","properties":"a:0:{}","default_value":null,"value":"Europe/Helsinki","normalizedValue":"Europe/Helsinki"},"country":{"id":"16","label":"Country","alias":"country","type":"country","group":"core","object":"lead","is_fixed":"1","properties":"a:0:{}","default_value":null,"value":"Greece","normalizedValue":"Greece"},"fax":{"id":"10","label":"Fax","alias":"fax","type":"tel","group":"core","object":"lead","is_fixed":"0","properties":"a:0:{}","default_value":null,"value":null,"normalizedValue":null},"preferred_locale":{"id":"17","label":"Preferred Locale","alias":"preferred_locale","type":"locale","group":"core","object":"lead","is_fixed":"1","properties":"a:0:{}","default_value":null,"value":null,"normalizedValue":null},"attribution_date":{"id":"20","label":"Attribution Date","alias":"attribution_date","type":"datetime","group":"core","object":"lead","is_fixed":"1","properties":"a:0:{}","default_value":null,"value":null,"normalizedValue":null},"attribution":{"id":"21","label":"Attribution","alias":"attribution","type":"number","group":"core","object":"lead","is_fixed":"1","properties":"a:2:{s:9:\"roundmode\";i:4;s:5:\"scale\";i:2;}","default_value":null,"value":null,"normalizedValue":null},"website":{"id":"22","label":"Website","alias":"website","type":"url","group":"core","object":"lead","is_fixed":"0","properties":"a:0:{}","default_value":null,"value":null,"normalizedValue":null},"agree_to_terms_of_use":{"id":"44","label":"Agree to terms of use","alias":"agree_to_terms_of_use","type":"boolean","group":"core","object":"lead","is_fixed":"0","properties":"a:2:{s:2:\"no\";s:2:\"No\";s:3:\"yes\";s:3:\"Yes\";}","default_value":"","value":null,"normalizedValue":null},"agree_to_receive_promotio":{"id":"45","label":"Agree to receive promotions","alias":"agree_to_receive_promotio","type":"boolean","group":"core","object":"lead","is_fixed":"0","properties":"a:2:{s:2:\"no\";s:2:\"No\";s:3:\"yes\";s:3:\"Yes\";}","default_value":"","value":null,"normalizedValue":null},"verification_pin":{"id":"47","label":"Verification PIN","alias":"verification_pin","type":"text","group":"core","object":"lead","is_fixed":"0","properties":"a:0:{}","default_value":null,"value":null,"normalizedValue":null},"interested_in_viber":{"id":"48","label":"Interested in Viber","alias":"interested_in_viber","type":"boolean","group":"core","object":"lead","is_fixed":"0","properties":"a:2:{s:2:\"no\";s:2:\"No\";s:3:\"yes\";s:3:\"Yes\";}","default_value":"","value":null,"normalizedValue":null},"interested_in_sms":{"id":"50","label":"Interested in SMS","alias":"interested_in_sms","type":"boolean","group":"core","object":"lead","is_fixed":"0","properties":"a:2:{s:2:\"no\";s:2:\"No\";s:3:\"yes\";s:3:\"Yes\";}","default_value":"","value":null,"normalizedValue":null}},"social":{"facebook":{"id":"23","label":"Facebook","alias":"facebook","type":"text","group":"social","object":"lead","is_fixed":"0","properties":"a:0:{}","default_value":null,"value":null,"normalizedValue":null},"foursquare":{"id":"24","label":"Foursquare","alias":"foursquare","type":"text","group":"social","object":"lead","is_fixed":"0","properties":"a:0:{}","default_value":null,"value":null,"normalizedValue":null},"instagram":{"id":"25","label":"Instagram","alias":"instagram","type":"text","group":"social","object":"lead","is_fixed":"0","properties":"a:0:{}","default_value":null,"value":null,"normalizedValue":null},"linkedin":{"id":"26","label":"LinkedIn","alias":"linkedin","type":"text","group":"social","object":"lead","is_fixed":"0","properties":"a:0:{}","default_value":null,"value":null,"normalizedValue":null},"skype":{"id":"27","label":"Skype","alias":"skype","type":"text","group":"social","object":"lead","is_fixed":"0","properties":"a:0:{}","default_value":null,"value":null,"normalizedValue":null},"twitter":{"id":"28","label":"Twitter","alias":"twitter","type":"text","group":"social","object":"lead","is_fixed":"0","properties":"a:0:{}","default_value":null,"value":null,"normalizedValue":null}},"personal":[],"professional":[]},"hasEditAccess":true,"custom":[{"attr":{"href":"/s/contacts/view/42279","data-toggle":"ajax","data-method":"POST"},"btnText":"mautic.core.details","iconClass":"fa fa-info-circle"}],"columsAliases":null} []`

Can anybody help me resolve this?
thank you
Dinos

**Comment by Aerendir** [comment]
There are two possible problems:

1. The files were not correctly updated (maybe for filesystem permissions issues)
2. The database is not in sync and so it doesn't generate correctly the variables needed by the template.

I don't know how to help you via GitHub issues :(

I can give you some guidance, but you have to be practical with command line (and I don't know if you are a developer).

Try to check the database schema is in sync:

    php bin/console doctrine:schema:validate

Check what the command outputs...

PS
Format the code clicking on the icon "<>" in the editor.

**Comment by dinaras** [comment]
Hello, thank you for your reply. I am cinfident in linux, but not an actual developer, more of a systems admin.

I did as you instructed and i got the following:
```
php bin/console doctrine:schema:validate

Mapping
-------

 [FAIL] The entity-class Mautic\DynamicContentBundle\Entity\DynamicContentLeadData mapping is invalid:
 * The association Mautic\DynamicContentBundle\Entity\DynamicContentLeadData#dynamicContent refers to the inverse side field Mautic\DynamicContentBundle\Entity\DynamicContent#id which is not defined as association.
 * The association Mautic\DynamicContentBundle\Entity\DynamicContentLeadData#dynamicContent refers to the inverse side field Mautic\DynamicContentBundle\Entity\DynamicContent#id which does not exist.


Database
--------

                                                                                                                        
 [ERROR] The database schema is not in sync with the current mapping file.                                              
                                                                                                                        


```
What should i try next?

**Comment by Aerendir** [comment]
Ok, so the first thing you should do is to keep in sync the database with the mapping: this means that entities set some fields and some relations, but they are not committed to the database.

The best thing you can do is to follow instructions here: https://github.com/mautic/mautic/issues/9129#issuecomment-707708797

Read also here: https://github.com/mautic/mautic/issues/9129#issuecomment-806565061

They are two comments from the same issue: read all the issue to understand what's going on.

You will have to run the command `php bin/console doctrine:migrations:migrate` two times: before running it the second time, edit the migration `Version20201120122846.php` as explained here: https://github.com/mautic/mautic/issues/10870#issuecomment-1045982049

**Obviously, before starting DO A BACKUP OF THE DATABASE: this entire procedure is highly risky!**

**Comment by dinaras** [comment]
Hello again,
Thank you for your detailed reply. I started doing what is proposed. So i get my database backed up and try to run 
`php bin/console doctrine:migrations:migrate`
It says 
`Your database version (5.5.60-MariaDB) does not support generated columns. Upgrade at least to 10.2.6 to get the speed improvements.
WARNING! You are about to execute a database migration that could result in schema changes and data loss. Are you sure you wish to continue? (y/n)n
Migration cancelled!
`
So i stop this script and set to update my database. After upgrading the database and rebooted server as i applied other updates aswell, everything started to work.
I dont know if i should run the above command now. 
My current mariadb version is now
mariadb.x86_64                       1:5.5.68-1.el7                  @base      
mariadb-libs.x86_64                  1:5.5.68-1.el7                  @base      
mariadb-server.x86_64                1:5.5.68-1.el7                  @base  
Using Centos as my operating system and nginx with php 7.4
Thank you

**Comment by Aerendir** [comment]
@dinaras , yes, you have to run it anyway: DO A BACKUP!

Your database currently is not in sync with the latest schema, so, also if for now Mautic is working, you will have troubles updating your installation.

DO A BACKUP, then run migrations, then follow the instructions I linked.

When you are finished, running `doctrine:schema:validate` should not output any error nor alert. Until then, you have a not in synch database.

**Comment by dinaras** [comment]
I am coming back to this issue as i have had some more time to look into it.
it appears that the problem in my case will go away, if mysql is restarted. Dont know yet for how long, and what triggers it to break. 
I also upgraded mariadb to 10.7 and run all proposed solutions above including upgrade, running queries in database proposed in some tickets. I even restored a couple of times but always i got to the same result.

Another thing i did not manage to solve is
`php bin/console doctrine:schema:validate

Mapping
-------

 [FAIL] The entity-class Mautic\DynamicContentBundle\Entity\DynamicContentLeadData mapping is invalid:
 * The association Mautic\DynamicContentBundle\Entity\DynamicContentLeadData#dynamicContent refers to the inverse side field Mautic\DynamicContentBundle\Entity\DynamicContent#id which is not defined as association.
 * The association Mautic\DynamicContentBundle\Entity\DynamicContentLeadData#dynamicContent refers to the inverse side field Mautic\DynamicContentBundle\Entity\DynamicContent#id which does not exist.


Database
--------

                                                                                                                        
 [OK] The database schema is in sync with the mapping files.  `

I have seen that you have created another ticket about the above problem. Will hopefully wait for a solution from that side now[ #10864 ](https://github.com/mautic/mautic/issues/10864)
Thank you very much for your help
Regards
Konstantinos

**Comment by Aerendir** [comment]
@dinaras , yes, the error about mapping is known and has to be fixed... If that one is the only error you get, then your mapping is valid and you should be able to use Mautic in its full potential without any issue.
